src/test/java/uk/gov/hmcts/reform/sscs/service/hmc/topic/ProcessMessageServiceTest.java

```diff
@@ -0,0 +1,276 @@
+    private SscsCaseData caseData;
+    private HearingGetResponse hearingGetResponse;
+    private HmcMessage hmcMessage;
+
+    @Mock
+    private HmcHearingApiService hmcHearingApiService;
+
+    @Mock
+    private CcdCaseService ccdCaseService;
+
+    @Mock
+    private CaseStateUpdateService caseStateUpdateService;
+
+    @InjectMocks
+    private ProcessMessageService processMessageService;
+
+    @BeforeEach
+    void setUp() throws GetHearingException, GetCaseException {
+        hearingGetResponse = HearingGetResponse.builder()
+                .hearingResponse(HearingResponse.builder().build())
+                .requestDetails(RequestDetails.builder().build())
+                .build();
+
+        given(hmcHearingApiService.getHearingRequest(HEARING_ID))
+                .willReturn(hearingGetResponse);
+
+        caseData = SscsCaseData.builder()
+                .state(UNKNOWN)
+                .ccdCaseId(String.valueOf(CASE_ID))
+                .build();
+
+        sscsCaseDetails = SscsCaseDetails.builder()
+                .data(caseData)
+                .build();
+
+        given(ccdCaseService.getCaseDetails(CASE_ID))
+                .willReturn(sscsCaseDetails);
+
+        hmcMessage = HmcMessage.builder()
+                .caseId(CASE_ID)
+                .hearingId(HEARING_ID)
+                .hearingUpdate(HearingUpdate.builder().build())
+                .build();
+    }
+
+    @DisplayName("When listing Status is Fixed and and HmcStatus is valid, "
+            + "updateListed and updateCaseData are called once")
+    @ParameterizedTest
+    @EnumSource(value = HmcStatus.class, names = {"LISTED", "UPDATE_SUBMITTED"})
+    void testUpdateListed(HmcStatus hmcStatus)
+            throws UpdateCaseException, GetHearingException, GetCaseException, InvalidIdException, InvalidHmcMessageException, InvalidMappingException, InvalidHearingDataException {
+        // given
+        hearingGetResponse.getHearingResponse().setListingStatus(FIXED);
+
+        hmcMessage.getHearingUpdate().setHmcStatus(hmcStatus);
+
+        givenHmcStatusUpdateCaseDataWillReturnSscsCaseDetails(sscsCaseDetails, hmcStatus);
+
+        // when
+        processMessageService.processEventMessage(hmcMessage);
+
+        // then
+        verify(caseStateUpdateService, times(1))
+                .updateListed(hearingGetResponse, hmcMessage, caseData);
+        verifyUpdateCaseDataCalledCorrectlyForHmcStatus(caseData, hmcStatus);
+    }
+
+
+
+    @DisplayName("When listing Status is not Fixed, updateListed and updateCaseData are not called")
+    @Test
+    void testUpdateListedNotFixed() throws Exception {
+        // given
+        hearingGetResponse.getHearingResponse().setListingStatus(DRAFT);
+        hmcMessage.getHearingUpdate().setHmcStatus(LISTED);
+
+        // when
+        processMessageService.processEventMessage(hmcMessage);
+
+        // then
+        verify(caseStateUpdateService, never()).updateListed(any(), any(), any());
```

Comment:
minor/optional: use `verifyZeroInteractions` instead? same for the others

Reply:
Done where possible